cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -281,7 +281,13 @@ func addDeviceModelAndVisitors(deviceModel *v1alpha1.DeviceModel, deviceProfile
 		propertyVisitor.PropertyName = pptv.PropertyName
 		propertyVisitor.ModelName = deviceModel.Name
 		if pptv.Modbus != nil {
-			propertyVisitor.Protocol = Modbus
+			if device.Spec.Protocol.Modbus != nil && device.Spec.Protocol.Modbus.TCP != nil {
```

Comment:
modbus mapper use protocol.protocol to check, so propertyVisitor.Protocol does not matter
![1](https://user-images.githubusercontent.com/12181200/88002679-a890f980-cb35-11ea-80e7-edc270d9bebc.png)